datafusion/sql/src/unparser/expr.rs

```diff
@@ -1799,6 +1915,33 @@ mod tests {
                 IntervalStyle::PostgresVerbose,
                 r#"INTERVAL '1 YEARS 7 MONS 0 DAYS 0 HOURS 0 MINS 0.00 SECS'"#,
             ),
+            (
+                interval_month_day_nano_lit(
```

Comment:
Is it valid to specify something like `1000000 SECOND`? Can we add a test case for overflowing intervals to test the logic.

Reply:
Confirm you can specify `1000000 SECOND` and it will be converted days/hours/etc. Added the following test, whether `1296000000` is `15000days x 24hours x 60 x 60`

```sql
(
    interval_month_day_nano_lit("1296000000 SECOND"),
    IntervalStyle::MySQL,
    r#"INTERVAL 15000 DAY"#,
),
```